peachjam/graph/ranker.py

```diff
@@ -73,11 +80,74 @@ def calculate_ranks(self):
 
     def publish_ranks(self):
         """Store ranks for each work."""
-        # TODO: only update those that have changed
-        # TODO: update elasticsearch directly?
-        works = {w.frbr_uri: w for w in Work.objects.all()}
+        works = {w.frbr_uri: {"work": w, "updated": False} for w in Work.objects.all()}
 
         for neo_work in NeoWork.nodes.all():
-            work = works[neo_work.frbr_uri]
-            work.ranking = neo_work.ranking
-            work.save(update_fields=["ranking"])
+            w = works[neo_work.frbr_uri]
+            # only update those that have changed
+            if w["work"].ranking != neo_work.ranking:
+                w["work"].ranking = neo_work.ranking
+                w["work"].save(update_fields=["ranking"])
+                w["work"]["updated"] = True
```

Comment:
This is ... odd. Rather than using a dict, maybe just use a list?

```python
updated = []
...
w = works[neo_work.frbr_uri]
if neo_work.ranking != w.ranking:
  w.ranking = neo_work.ranking
  w.save(updated_fields=["ranking"])
  updated.add(w)

self.update_index([w.frbr_uri for w in updated])
```

Reply:
changed :heavy_check_mark: